pkgs/build-support/setup-hooks/wrap-apps-hook/wrap-apps-hook.sh

```diff
@@ -0,0 +1,79 @@
+# shellcheck shell=bash
+appsWrapperArgs=()
+
+# Inherit arguments given in mkDerivation
+appsWrapperArgs=( ${appsWrapperArgs-} )
+
+makeAppWrapper() {
+    local original="$1"
+    local wrapper="$2"
+    shift 2
+    makeWrapper "$original" "$wrapper" "${appsWrapperArgs[@]}" "$@"
+}
+
+wrapApp() {
+    local program="$1"
+    shift 1
+    wrapProgram "$program" "${appsWrapperArgs[@]}" "$@"
+}
+
+wrapAppsHook() {
+    # skip this hook when requested
+    [ -z "${dontWrapApps-}" ] || return 0
+
+    # guard against running multiple times (e.g. due to propagation)
+    [ -z "$wrapAppsHookHasRun" ] || return 0
+    wrapAppsHookHasRun=1
+
+    local targetDirs =( "$prefix/bin" "$prefix/sbin" "$prefix/libexec"  )
+    echo "wrapping applications in ${targetDirs[@]}"
+
+    for targetDir in "${targetDirs[@]}"
+    do
+        [ -d "$targetDir" ] || continue
+
+        find "$targetDir" ! -type d -executable -print0 | while IFS= read -r -d '' file
+        do
+            # TODO: further checks
+            # Following is done e.g. for Qt
+            #
+            # patchelf --print-interpreter "$file" >/dev/null 2>&1 || continue
```

Comment:
This is the Qt check I mentioned. In my opinion we should wrap scripts as well. If there are scripts or simply executables to exclude, we should consider supporting an include and exclude list.

Reply:
Yup, we should wrap those too – we had issues where PyQt programs were not properly wrapped.